internal/pkg/service/buffer/api/service/import.go

```diff
@@ -38,90 +34,43 @@ func (s *service) Import(d dependencies.ForPublicRequest, payload *buffer.Import
 		}
 	}
 
-	data, err := parseRequestBody(payload.ContentType, reader)
+	body, err := receive.ReadBody(bodyReader)
 	if err != nil {
-		return err
+		return errors.Errorf(`cannot read request body: %w`, err)
 	}
 
-	importCtx := column.NewImportCtx(data, header, ip)
-	receivedAt := time.Now()
-
-	errs := errors.NewMultiError()
-	for _, e := range receiver.Exports {
-		// nolint:godox
-		// TODO get sliceID and fileID + use in stats.Notify
-		fileKey := key.FileKey{ExportKey: e.ExportKey, FileID: key.FileID(receivedAt)}
-		sliceKey := key.SliceKey{FileKey: fileKey, SliceID: key.SliceID(receivedAt)}
+	receiveCtx := receivectx.New(ctx, d.Clock().Now(), d.RequestClientIP(), d.RequestHeader(), body)
+	errs := errors.NewMultiErrorNoTrace()
+	for _, export := range receiver.Exports {
+		// Format CSV row
+		csvRow, err := receive.FormatCSVRow(receiveCtx, export)
+		if err != nil {
+			// Wrap error with export ID
+			err = errors.PrefixErrorf(err, `failed to format record for export "%s"`, export.ExportID)
 
-		csv := make([]string, 0)
-		for _, c := range e.Mapping.Columns {
-			csvValue, err := c.CsvValue(importCtx)
-			if err != nil {
-				return err
+			// Convert FormatCSVRow error to the BadRequestError, if it doesn't have a specific HTTP code
+			if HTTPCodeFrom(err) == http.StatusInternalServerError {
+				err = NewBadRequestError(err)
 			}
-			csv = append(csv, csvValue)
-		}
 
-		record := key.NewRecordKey(sliceKey, receivedAt)
-		err = str.CreateRecord(ctx, record, csv)
-		if err != nil {
-			errs.AppendWithPrefixf(err, `failed to create record for export "%s"`, e.ExportID)
+			errs.Append(err)
+			continue
 		}
 
-		size := uint64(0)
-		for _, column := range csv {
-			size += uint64(len(column))
+		// Persist record
+		sliceKey := export.OpenedSlice.SliceKey
+		recordKey := key.NewRecordKey(sliceKey, time.Now())
+		if err := str.CreateRecord(ctx, recordKey, csvRow); err != nil {
+			errs.AppendWithPrefixf(err, `failed to persist record for export "%s"`, export.ExportID)
+			continue
 		}
-		stats.Notify(sliceKey, size)
-	}
 
-	return errs.ErrorOrNil()
-}
-
-func parseRequestBody(contentType string, reader io.ReadCloser) (res *orderedmap.OrderedMap, err error) {
-	if !isContentTypeForm(contentType) && !regexp.MustCompile(`^application/([a-zA-Z0-9\.\-]+\+)?json$`).MatchString(contentType) {
-		return nil, NewUnsupportedMediaTypeError(errors.New(
-			"Supported media types are application/json and application/x-www-form-urlencoded.",
-		))
+		// Update statistics
+		s.stats.Notify(sliceKey, uint64(len(csvRow)))
```

Comment:
Uz sa spravne reportuje do statistik sirka celeho riadku, predtym sa to pocitalo trocha zvlastne po stlpcoch.